### PR TITLE
netcore/feature/msdi

### PR DESCRIPTION
**Pull request by rustybox**
LightInject gone, IRegister gone, IFactory gone
Good start for #8653

### CODE REVIEW

**Comment by rustybox** [comment]
```
Composing\LightInject\LightInjectContainer.cs(5,30): error CS0234: The type or namespace name 'LightInject' does not exist in the namespace 'Umbraco.Core.Composing'  
Composing\LightInject\LightInjectContainer.cs(12,56): error CS0234: The type or namespace name 'LightInject' does not exist in the namespace 'Umbraco.Core.Composing'  
Composing\LightInject\LightInjectContainer.cs(30,30): error CS0115: 'LightInjectContainer.EnablePerWebRequestScope()': no suitable method found to override  
Composing\LightInject\LightInjectContainer.cs(38,30): error CS0115: 'LightInjectContainer.ConfigureForWeb()': no suitable method found to override  
Composing\Current.cs(49,23): error CS0246: The type or namespace name 'IFactory' could not be found  
CompositionExtensions.cs(64,90): error CS0246: The type or namespace name 'IFactory' could not be found  
CompositionExtensions.cs(95,83): error CS0246: The type or namespace name 'IFactory' could not be found  
Composing\Current.cs(44,24): error CS0246: The type or namespace name 'IFactory' could not be found  
Mvc\ContainerControllerFactory.cs(10,31): error CS0246: The type or namespace name 'IFactory' could not be found  
Mvc\ContainerControllerFactory.cs(12,48): error CS0246: The type or namespace name 'IFactory' could not be found  
UmbracoApplicationBase.cs(128,27): error CS0246: The type or namespace name 'IRegister' could not be found  
UmbracoApplicationBase.cs(31,17): error CS0246: The type or namespace name 'IFactory' could not be found  
```

These seem pretty reasonable :D

What's the deal with Umbraco.Web, I've been focusing on those with a TFW of netstandard and netcoreapp

**Comment by bergmania** [comment]
@rustybox..  I think it is okay to just comment out the code in the `Umbraco.Web` project, so the entire solution still builds.

**Comment by rustybox** [comment]
If you look through the first 6 commits and you think to yourself, "He hasn't cottoned on to RegisterFor and UniqueFor" being for named registrations which aren't in MS DI, I get it now, starting to workaround where possible.

**Comment by rustybox** [comment]
It builds! 

Can this get merged @bergmania?

**Comment by bergmania** [comment]
@rustybox, it can't be merged as long as tests are not passing.  When tests are passing we will also need to review everything :)

**Comment by rustybox** [comment]
Oh, that's pretty much what I was getting at with my asking what the deal was with the 472 projects.

**Comment by Shazwazza** [comment]
@rustybox Just posting my reply on the main thread for reference here https://github.com/umbraco/Umbraco-CMS/issues/8653#issuecomment-691838771 let me know what you think.

**Review by Shazwazza — CHANGES_REQUESTED** [review]
I've left some notes inline about a few things. I think it will be good as part of this PR to create a list of follow up notes for what we might need to investigate and change (like the TODOs). The tests may have all run but the website does not work. We'll need to have the website actually run/boot before we can merge. When running it I get lots of errors about services not being able to be resolved.  As an example, the mapper objects like `ConsentMapper`. This takes in a ConcurrentDictionary as a ctor argument but it cannot resolve that. TBH I actually don't know how this currently works with the MapperCollectionBuilder since each mapper there has this ConcurrentDictionary passed in.

With the latest netcore changes there's also some file conflicts that need to be  resolved but hopefully those are all easy to do.

File: src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs
Comment (rustybox) [reply src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs:23]: PhysicalFileSystem, there was a registration with a factory method that set it up as such

File: src/Umbraco.Core/Composing/CollectionBuilderBase.cs
Comment (rustybox) [reply src/Umbraco.Core/Composing/CollectionBuilderBase.cs:121]: I just scratched my head and gave up when I saw the dicts, are the mappers supposed to share a ConcurrentDictionary or have one per each? 

Regardless of the answer to the above, it's a bit too generic of a resolve, perhaps needs a wrapper to make things more obvious, an IMapperThingerDictionary

File: src/Umbraco.Core/IO/PhysicalFileSystem.cs
Comment (rustybox) [reply src/Umbraco.Core/IO/PhysicalFileSystem.cs:14]: Yep no worries

File: src/Umbraco.Core/Composing/CollectionBuilderBase.cs
Comment (rustybox) [reply src/Umbraco.Core/Composing/CollectionBuilderBase.cs:121]: > This takes in a ConcurrentDictionary as a ctor argument but it cannot resolve that. TBH I actually don't know how this currently works with the MapperCollectionBuilder since each mapper there has this ConcurrentDictionary passed in.

Do you know if the dict should be shared among all mappers or one per each, perhaps a wrapper around the dependency would be good, it's a bit strange to resolve a generic collection from the container.

Comment (Shazwazza) [reply src/Umbraco.Core/Composing/CollectionBuilderBase.cs:121]: Yeah the Mappers with the ConcurrentDictionary param is strange. I 100% agree that it should have an actual object/interface injected instead that perhaps just contains the concurrent dictionary instance, at least it would make it clear as to what is being injected. I had a look into v8 and the way this is currently being injected is here: https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Persistence/Mappers/MapperCollectionBuilder.cs#L29 where it manually creates the instance and passes in `_maps`. So yes it is the same instance passed in to all of them.

File: src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs
Comment (rustybox) [reply src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs:23]: I think at one point I thought I would keep FileTreesController depending on IFileSystem and wire it up to resolve IPhysicalFileSystem in the container.

Would a FileTreesController ever benefit from using an alternative implementation of IFileSystem?

**Comment by rustybox** [comment]
> We'll need to have the website actually run/boot before we can merge.

Is that the legacy site or Umbraco.Web.UI.NetCore or both?

**Comment by bergmania** [comment]
> Is that the legacy site or Umbraco.Web.UI.NetCore or both?

Only `Umbraco.Web.UI.NetCore`

**Comment by rustybox** [comment]
Awesome, if I want to sync with upstream, which branch should I merge, netcore/netcore or netcore/dev?

or phrased alternatively, what's the purpose of netcore/netcore?

**Comment by bergmania** [comment]
> Awesome, if I want to sync with upstream, which branch should I merge, netcore/netcore or netcore/dev?
> 
> or phrased alternatively, what's the purpose of netcore/netcore?

- `netcore/dev` is where we have changed all the underlying libraries to be netstandard, so we can still test the website and backoffice works.

- `netcore/netcore` is where we have migrated the backoffice, but still not the website part.  As soon as the website is also working, we will merge this down to `netcore/dev`

I think this is okay if it is just working in `netcore/netcore`.

**Comment by rustybox** [comment]
#progress 

a_form_has_appeared

**Comment by rustybox** [comment]
I have WIP on merging upstream, going to have to come back to it in a day or two though.

File: src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs
Comment (Shazwazza) [reply src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs:23]: Nope in fact its coded explicitly to use the physical file system in v8 https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web/Trees/FilesTreeController.cs#L11 this change must have been a refactoring change in netcore. There should never really be a reason to not have as the PhysicalFileSystem ... except for unit tests  :)

**Comment by rustybox** [comment]
I'm nuking this and starting fresh from netcore/netcore @ head

I'm two nights down rabbit holes and comparing to netcore/netcore is a pain

I think I'd like to iterate slower, lose LightInject without losing IFactory etc just yet.

**Comment by JimBobSquarePants** [comment]
Oh dear.... maybe the core team can react a shade faster? This is a lot of good wasted community work.

**Comment by Shazwazza** [comment]
Another PR was made here https://github.com/umbraco/Umbraco-CMS/pull/9247 this just a little too behind but the new approach will be more manageable, nice work 👍

**Pull request closed without merging**
